pallets/runtime/tests/src/settlement_test.rs

```diff
@@ -3066,3 +3083,371 @@ fn multiple_custodian_settlement() {
+
+            // ------------ Setup mercat
+            let token_name = b"ACME";
+            let ticker = Ticker::try_from(&token_name[..]).unwrap();
+            // Create an account for Dave and mint 10,000,000 tokens to ACME
+            let tx_id = create_account_and_mint_token(
+                dave, // owner of ACME
+                dave_did,
+                1_000_000, // total supply
+                token_name.to_vec(),
+                0,          // transaction id: not important in this test
+                [10u8; 32], // seed. TODO: replace all these seeds with rng
+            );
+
+            // Create an account for Alice
+            let tx_id = tx_id + 1;
+            let (alice_secret_account, alice_account_id) = init_account(
+                tx_id, [11u8; 32], // seed. TODO: replace all these seeds with rng
+                token_name, alice,
+            );
+            let alice_public_account =
+                ConfidentialAsset::mercat_accounts(alice_did, alice_account_id.clone())
+                    .to_mercat::<TestStorage>()
+                    .unwrap();
+            let alice_encrypted_init_balance =
+                ConfidentialAsset::mercat_account_balance(alice_did, alice_account_id)
+                    .to_mercat::<TestStorage>()
+                    .unwrap();
+            // Create an account for Bob
+            let tx_id = tx_id + 1;
+            let (bob_secret_account, bob_account_id) = init_account(
+                tx_id, [12u8; 32], // seed. TODO: replace all these seeds with rng
+                token_name, bob,
+            );
+            let bob_public_account = ConfidentialAsset::mercat_accounts(bob_did, bob_account_id)
+                .to_mercat::<TestStorage>()
+                .unwrap();
+            // Create a mediator account for Charlie
+            let tx_id = tx_id + 1;
+            let (charlie_secret_account, charlie_account_id) = init_account(
+                tx_id, [13u8; 32], // seed. TODO: replace all these seeds with rng
+                token_name, charlie,
+            );
+            let charlie_public_account =
+                ConfidentialAsset::mercat_accounts(charlie_did, charlie_account_id)
+                    .to_mercat::<TestStorage>()
+                    .unwrap();
+
+            // Mediator creates a venue
+            let venue_counter = Settlement::venue_counter();
+            assert_ok!(Settlement::create_confidential_venue(
+                Origin::signed(charlie),
+                VenueDetails::default(),
+                vec![charlie],
+                VenueType::Other
+            ));
+
+            // Mediator creates an instruction
+            let instruction_counter = Settlement::instruction_counter();
+
+            //// Provide scope claim to sender and receiver of the transaction.
+            //provide_scope_claim_to_multiple_parties(&[alice_did, bob_did], ticker, dave); // TODO: I think we decided not to do this as it would leak the ticker name
```

Comment:
I think the plan was to have the mediator check the scope claims and not leak the ticker name, is that doable?

Reply:
Not sure, will create another task for it.

Reply:
https://polymath.atlassian.net/browse/CRYP-172